### ISSUE TITLE
checkbox: is it a11y?

### ISSUE THREAD

**Issue by danethurber** [pluralsight/design-system#1358]
a question was brought up about the a11y of the checkbox because it fails with test-library -> https://pluralsight.slack.com/archives/C70HPSJPP/p1603722908130800

looking at the code i think we can/should improve the a11y

some guidelines:
https://www.a11ymatters.com/pattern/checkbox/#custom-checkbox-design

**Comment by jaketrent** [comment]
Team triage: htmlFor->id between label and input as in the "Using Inline SVG" section of the a11ymatters article.
